server/middlewares/stats.js

```diff
@@ -48,6 +51,36 @@ const incrCountersBy = (key, value) => {
   client.incrby(`${key}:daily:${year}-${month}-${day}`, value)
 }
 
+const initVisitor = (trackingID, userID) => {
+  return ua({
+    tid: trackingID,
+    cid: userID,
+    strictCidFormat: false,
+  })
+}
+
+const logActivity = (type, ...args) => {
```

Comment:
I would split this into two separate functions - trackPageView and trackEvent (and rename `log` to `track` since `log` may be misleading) and I would move the error handling to trackVisits/trackTxSubmissions. In trackVisits we probably do not want to add the ip to `knownIps` if tracking the visit fails, so it makes sense to handle the error there.

In general, it makes sense to handle the error as high as possible.